crates/epoch-snark/src/gadgets/epochs.rs

```diff
@@ -178,14 +187,62 @@ impl ValidatorSetUpdate<Bls12_377> {
                 self.hash_helper.is_none(), // generate constraints in BW6_761 if no helper was provided
             )?;
 
+            let index_bit = YToBitGadget::<Parameters>::is_eq_zero(
+                &mut cs.ns(|| format!("is index {} zero", i)),
+                &constrained_epoch.index,
+            )?.not();
+
             // Update the pubkeys for the next iteration
-            previous_epoch_index = constrained_epoch.index;
-            previous_pubkey_vars = constrained_epoch.new_pubkeys;
-            previous_max_non_signers = constrained_epoch.new_max_non_signers;
+            previous_epoch_index = FrGadget::conditionally_select(
```

Comment:
What's the point of updating this value if it's never used anywhere later? Same for `previous_max_non_signers` and `previous_pub_key_vars`.

Reply:
All of these are used when they're passed down to inner functions.

Reply:
I don't follow. When I searched the file I only saw these values used to update themselves. Didn't see them used in any of the conditional logic nor are they returned.

Reply:
All of them are passed to `epoch.constrain`.

Reply:
Thanks, I'm blind.